src/python_testing/TC_DeviceBasicComposition.py

```diff
@@ -338,6 +339,41 @@ class RequiredMandatoryAttribute:
                         success = False
                         continue
 
+        # Validate there are attributes in the global range that are not in the required list
+        allowed_globals = [a.id for a in ATTRIBUTES_TO_CHECK]
+        # also allow event list because it's not disallowed
+        event_list_id = 0xFFFA
+        allowed_globals.append(event_list_id)
+        global_range_min = 0x0000_F000
+        mei_range_min = 0x0001_0000
+        attribute_list_id = 0xFFFB
+        for endpoint_id, endpoint in self.endpoints_tlv.items():
+            for cluster_id, cluster in endpoint.items():
+                globals = [a for a in cluster[attribute_list_id] if a >= global_range_min and a < mei_range_min]
+                unexpected_globals = sorted(list(set(globals) - set(allowed_globals)))
+                for unexpected in unexpected_globals:
+                    location = AttributePathLocation(endpoint_id=endpoint_id, cluster_id=cluster_id, attribute_id=unexpected)
+                    self.record_error(self.get_test_name(), location=location,
+                                      problem=f"Unexpected global attribute {unexpected} in cluster {cluster_id}", spec_location="Global elements")
+                    success = False
+
+        # validate that all the returned attributes in the standard clusters contain only known attribute ids
+        for endpoint_id, endpoint in self.endpoints_tlv.items():
+            for cluster_id, cluster in endpoint.items():
+                if cluster_id not in chip.clusters.ClusterObjects.ALL_ATTRIBUTES:
+                    # Skip clusters that are not part of the standard generated corpus (e.g. MS clusters)
+                    continue
+                standard_attributes = [a for a in cluster[attribute_list_id] if a < global_range_min]
```

Comment:
This isn't right.  The top of the non-global standard attribute range is 0x4FFF, not 0xEFFF, so that test should be `a < 0x5000` or `a <= 0x4FFF` or something like that, perhaps with better naming.

Reply:
It ends up being the same thing because the spec will never define a standard value in the range above 0x4FFF, but I can make the check separate to make that obvious

Reply:
separated.